app/src/main/java/com/android/voyageur/ui/overview/Overview.kt

```diff
@@ -338,17 +412,39 @@ fun DisplayParticipants(trip: Trip, userViewModel: UserViewModel) {
   }
 }
 
-// Helper function to convert Timestamp to String format.
+/** Helper function to convert Timestamp to String format. */
 fun Timestamp.toDateString(): String {
   val sdf = java.text.SimpleDateFormat("MMM dd yyyy", java.util.Locale.getDefault())
   return sdf.format(this.toDate())
 }
 
-// Helper function to generate the correct string
+/** Helper function to generate the correct string for the number of participants */
 fun generateParticipantString(numberOfParticipants: Int): String {
   return when (numberOfParticipants) {
     0 -> "No participants."
     1 -> "1 Participant:"
     else -> "$numberOfParticipants Participants:"
   }
 }
+/**
+ * Opens the default calendar app on the device with pre-filled details for a new event.
+ *
+ * This method creates an `Intent` to insert a calendar event using the trip's details such as name,
+ * description, start time, and end time.
+ *
+ * @param context The context used to launch the intent.
+ * @param trip The trip data used to populate the calendar event details.
+ */
+internal fun openGoogleCalendar(context: Context, trip: Trip) {
+  // Create an intent to insert into the calendar app
+  val intent =
+      Intent(Intent.ACTION_INSERT)
+          .setData(CalendarContract.Events.CONTENT_URI)
+          .putExtra(CalendarContract.Events.TITLE, trip.name)
+          .putExtra(CalendarContract.Events.DESCRIPTION, trip.description)
+          .putExtra(CalendarContract.EXTRA_EVENT_BEGIN_TIME, trip.startDate.toDate().time)
+          .putExtra(CalendarContract.EXTRA_EVENT_END_TIME, trip.endDate.toDate().time)
+          .putExtra(CalendarContract.Events.EVENT_TIMEZONE, TimeZone.getDefault().id)
+  // Start Activity and open calendar
+  context.startActivity(intent)
+}
```

Comment:
When launching the calendar intent, the code assumes that a calendar-capable app is available. Consider checking for a suitable activity using, for example, resolveActivity() before calling startActivity(intent). If none is found, you should handle the situation—perhaps by showing a Toast message to inform the user that no compatible application is installed. Additionally, wrapping the startActivity call in a try/catch block to handle ActivityNotFoundException is a good safety measure.

Reply:
Also, openGoogleCalendar() uses trip.startDate and trip.endDate directly without verifying their validity. Since they are sourced from user input or a network call, they might be null or invalid. To prevent crashes or unexpected behavior, ensure that Trip objects always contain valid timestamps or provide a fallback when data is missing.

Reply:
> When launching the calendar intent, the code assumes that a calendar-capable app is available. Consider checking for a suitable activity using, for example, resolveActivity() before calling startActivity(intent). If none is found, you should handle the situation—perhaps by showing a Toast message to inform the user that no compatible application is installed. Additionally, wrapping the startActivity call in a try/catch block to handle ActivityNotFoundException is a good safety measure.

Thanks for the suggestion! I think you're right, I just handled the exception. I didn't initially add this because the android comes with a calendar downloaded so I thought it unlikely for a user to disable the app and then try to add the trip to the calendar, but handling it solves the issue more gracefully and prevents our app from crashing.

Reply:
> Also, openGoogleCalendar() uses trip.startDate and trip.endDate directly without verifying their validity. Since they are sourced from user input or a network call, they might be null or invalid. To prevent crashes or unexpected behavior, ensure that Trip objects always contain valid timestamps or provide a fallback when data is missing.

Thanks! I see your point, but when you add a trip to the calendar, you're selecting it directly from overview (which means it already has startDate and endDate as valid timestamps, else it couldn't have been created). However, I understand being on the safe side, so if you think it's really necessary I can add it. Let me know what you think!

Reply:
> > When launching the calendar intent, the code assumes that a calendar-capable app is available. Consider checking for a suitable activity using, for example, resolveActivity() before calling startActivity(intent). If none is found, you should handle the situation—perhaps by showing a Toast message to inform the user that no compatible application is installed. Additionally, wrapping the startActivity call in a try/catch block to handle ActivityNotFoundException is a good safety measure.
> 
> Thanks for the suggestion! I think you're right, I just handled the exception. I didn't initially add this because the android comes with a calendar downloaded so I thought it unlikely for a user to disable the app and then try to add the trip to the calendar, but handling it solves the issue more gracefully and prevents our app from crashing.

Thank you for considering the suggestion about handling exceptions and checking for compatible apps.

Reply:
> > Also, openGoogleCalendar() uses trip.startDate and trip.endDate directly without verifying their validity. Since they are sourced from user input or a network call, they might be null or invalid. To prevent crashes or unexpected behavior, ensure that Trip objects always contain valid timestamps or provide a fallback when data is missing.
> 
> Thanks! I see your point, but when you add a trip to the calendar, you're selecting it directly from overview (which means it already has startDate and endDate as valid timestamps, else it couldn't have been created). However, I understand being on the safe side, so if you think it's really necessary I can add it. Let me know what you think!

Since the creation process guarantees valid timestamps, then this might indeed be redundant. However, ensuring safety through validation could still be worthwhile, especially if there’s ever a future change in the workflow that could introduce variability. It might not be critical right now, but adding it as a safeguard could prevent issues down the line. That said, I trust your judgment here, so feel free to decide!

Reply:
I will leave them like this because I've tried to implement them in try-catch blocks,  but this reduces our line coverage under 80% , and I can't provide tests for them (since I have no way to create a trip with an invalid startDate/endDate to make the method .toDate() or .time to fail and even if I did, I could not test because they are throwing toasts in case of an exception). As this is not critical for now, if you agree, I would leave it like the original implementation.

Reply:
I agree that this is a good solution for now. It’s worth noting that this is not the original implementation, but the current one in the branch, which includes the modification to handle ActivityNotFoundException with the toast fallback, which I believe to be quite important. For now, maintaining the code coverage as high as possible is important, so we can keep it as is. We can revisit the other aspects in the final sprint when the app is closer to release to manage these better.